homeassistant/components/tractive/sensor.py

```diff
@@ -0,0 +1,178 @@
+        self._trackable = trackable
+        self._tracker_id = tracker_details["_id"]
+        self._attr_unique_id = unique_id
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, tracker_details["_id"])},
+            "name": f"Tractive ({tracker_details['_id']})",
+            "manufacturer": "Tractive GmbH",
+            "sw_version": tracker_details["fw_version"],
+            "model": tracker_details["model_number"],
+        }
+        self.entity_description = description
+
+    @callback
+    def handle_server_unavailable(self):
+        """Handle server unavailable."""
+        self._attr_available = False
+        self.async_write_ha_state()
+
+
+class TractiveHardwareSensor(TractiveSensor):
+    """Tractive hardware sensor."""
+
+    def __init__(self, user_id, trackable, tracker_details, unique_id, description):
+        """Initialize sensor entity."""
+        super().__init__(user_id, trackable, tracker_details, unique_id, description)
+        self._attr_name = f"{self._tracker_id} {description.name}"
+
+    @callback
+    def handle_hardware_status_update(self, event):
+        """Handle hardware status update."""
+        self._attr_native_value = event[self.entity_description.key]
+        self._attr_available = True
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self):
+        """Handle entity which will be added."""
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{TRACKER_HARDWARE_STATUS_UPDATED}-{self._tracker_id}",
+                self.handle_hardware_status_update,
+            )
+        )
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{SERVER_UNAVAILABLE}-{self._user_id}",
+                self.handle_server_unavailable,
+            )
+        )
+
+
+class TractiveActivitySensor(TractiveSensor):
+    """Tractive active sensor."""
+
+    def __init__(self, user_id, trackable, tracker_details, unique_id, description):
+        """Initialize sensor entity."""
+        super().__init__(user_id, trackable, tracker_details, unique_id, description)
+        self._attr_name = f"{trackable['details']['name']} {description.name}"
+
+    @callback
+    def handle_activity_status_update(self, event):
+        """Handle activity status update."""
+        self._attr_native_value = event[self.entity_description.key]
+        self._attr_extra_state_attributes = {
+            attr: event[attr] if attr in event else None
+            for attr in self.entity_description.attributes
+        }
+        self._attr_available = True
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self):
+        """Handle entity which will be added."""
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{TRACKER_ACTIVITY_STATUS_UPDATED}-{self._trackable['_id']}",
```

Comment:
Is this correct? `tracker_details["_id"]` is used in the device tracker entities

Reply:
Yes, `activity update` message doesn't contain `tracker_id` but contains `pet_id` (`self._trackable['_id']` is `pet_id`) which is used as `id` [here](https://github.com/home-assistant/core/blob/02e16360b0032dce589a2969f345715395fa1a6c/homeassistant/components/tractive/__init__.py#L153).